### PR TITLE
Graph buffering

### LINKED ISSUE #166: Requesting more data in the Detail-Graph
If you have a nice detail-graph in front of you and notice some suspicious behaviour near one or both of the sides (or just want to get some larger overview), you have to manually adjust the start or end date.
Even worse, if you want to symmetrically expand the display, e.g. fetch a week more on both sides, you need to adjust *both* of the dates after each other.

What are nice ways to deal with this? Some random ideas:
- Have a "+ 1 week" button that symmetrically extends the range on both sides
- Zooming out using the mouse wheel when the graph is fully zoomed out expands the range symmetrically on both sides. I have no idea how well - or even if - this can be implemented, but it has some pros and cons:
  - It seems reasonable intuitive. You are fully zoomed out and zoom out further, so you are moving the boundaries
  - If you just want to zoom out and quickly scroll the mouse wheel, you might accidentally expand the window and trigger a refetch
- Panning the graph (Shift+Dragging) beyond the edge causes the start date to be expanded by a week in this direction after you let go of the mouse (Not sure if you can implement that in echarts, but it might be possible)
- Use a https://vuetifyjs.com/en/components/date-pickers/#range for the start / end date which might make these adjustments more tolerable
- Make a custom widget similar to the sliders below the graph where you can interactively select the range you want
- Probably many better and different ideas I missed

### PR DESCRIPTION
**Pull request by oudemia**
Until now, both graphs were limited to the requested time span.
This PR introduces the possibility to pan and zoom out outside to the current graph boundaries in both graphs.
When moving outside of the current graph, new data is automatically loaded.
This PR closes #166 .

### CODE REVIEW

File: frontend/src/components/graphs/DytailGraph.vue [hunk 1]
```diff
@@ -70,15 +106,162 @@ export default class DytailGraph extends Vue {
...
+   * no more than two days -> hh:mm \n mm-dd
+   * otherwise             -> mm-dd \n yyyy
+   * @param d: date to format
+   * @param start: start value of x axis (required by dygraph)
+   * @param end: end value of x axis (required by dygraph)
+   * @private
+   */
   private xAxisFormatter(d: Date, [start, end]: [number, number]): string {
-    const dateString: string = d.getDate() + '-' + (d.getMonth() + 1)
+    const dateString: string = d.getMonth() + 1 + '-' + d.getDate()
```
Comment (I-Al-Istannen) [inline frontend/src/components/graphs/DytailGraph.vue:264]: I think I'd add clarifying parens there

File: frontend/src/components/graphs/DytailGraph.vue [hunk 2]
```diff
@@ -295,126 +449,123 @@ export default class DytailGraph extends Vue {
...
-  @Watch('dayEquidistant')
-  private up() {
-    const data: number[][] = []
-
-    // One array entry = #dimensions data points per commit
-    // one array entry has the form [x-val, dim1, dim2, ...]
-    for (let i = 0; i < this.datapoints.length; i++) {
-      data[i] = [this.datapoints[i].positionDate.getTime()]
+  /**
+   * draw a little flag to merk the reference data point, if specified and in the
```
Comment (I-Al-Istannen) [inline frontend/src/components/graphs/DytailGraph.vue:453]: mark

File: frontend/src/components/graphs/DytailGraph.vue [hunk 3]
```diff
@@ -295,126 +449,123 @@ export default class DytailGraph extends Vue {
...
-      includeZero: this.beginYAtZero,
-      rangeSelectorPlotLineWidth: this.selectorLineWidth,
-      rangeSelectorAlpha: this.selectorAlpha
-    } as RealOptions)
+  // <!--<editor-fold desc="CUSTOM GRAPH INTERACTIONS">-->
 
-    this.drawMarkers()
-  }
+  /**
+   * what to when interacting with a point in the graph:
```
Comment (I-Al-Istannen) [inline frontend/src/components/graphs/DytailGraph.vue:479]: what to do?

**Review by I-Al-Istannen — APPROVED** [review]

File: frontend/src/components/graphs/DytailGraph.vue (hunk 1 above)
Comment (oudemia) [reply frontend/src/components/graphs/DytailGraph.vue:264]: prettier doesn't let me keep them... :/

Comment (I-Al-Istannen) [reply frontend/src/components/graphs/DytailGraph.vue:264]: Heh. Alright :D

**Pull request merged**